flink-table-store-connector/src/main/java/org/apache/flink/table/store/connector/TableStoreCompactJob.java

```diff
@@ -0,0 +1,147 @@
+import org.apache.flink.configuration.ReadableConfig;
+import org.apache.flink.core.execution.JobClient;
+import org.apache.flink.core.fs.Path;
+import org.apache.flink.streaming.api.datastream.DataStreamSource;
+import org.apache.flink.streaming.api.environment.StreamExecutionEnvironment;
+import org.apache.flink.table.data.RowData;
+import org.apache.flink.table.store.CoreOptions;
+import org.apache.flink.table.store.connector.sink.CompactorSinkBuilder;
+import org.apache.flink.table.store.connector.source.CompactorSourceBuilder;
+import org.apache.flink.table.store.connector.utils.StreamExecutionEnvironmentUtils;
+import org.apache.flink.table.store.table.FileStoreTable;
+import org.apache.flink.table.store.table.FileStoreTableFactory;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/** Stand-alone compact job for table store. */
+public class TableStoreCompactJob {
+
+    public static void main(String[] args) throws Exception {
+        if (args.length < 1 || args.length > 4) {
+            printHelp();
+            System.exit(1);
+        }
+
+        Path location;
+        if (args.length <= 2) {
+            location = new Path(args[0]);
+        } else {
+            location = new Path(new Path(args[0], args[1] + ".db"), args[2]);
+        }
+
+        String partitionsSpecString = null;
+        if (args.length == 2) {
+            partitionsSpecString = args[1];
+        } else if (args.length == 4) {
+            partitionsSpecString = args[3];
+        }
+
+        StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment();
+
+        runCompactJob(location, partitionsSpecString, env);
+    }
+
+    @VisibleForTesting
+    public static JobClient runCompactJob(
+            Path location, String partitionsSpecString, StreamExecutionEnvironment env)
+            throws Exception {
+        Configuration tableOptions = new Configuration();
+        tableOptions.set(CoreOptions.PATH, location.toString());
+        tableOptions.set(CoreOptions.WRITE_COMPACTION_SKIP, false);
+        FileStoreTable table = FileStoreTableFactory.create(tableOptions);
+
+        List<Map<String, String>> specifiedPartition = parsePartitionSpec(partitionsSpecString);
+
+        ReadableConfig conf = StreamExecutionEnvironmentUtils.getConfiguration(env);
+        boolean isStreaming =
+                conf.get(ExecutionOptions.RUNTIME_MODE) == RuntimeExecutionMode.STREAMING;
+
+        DataStreamSource<RowData> source =
+                new CompactorSourceBuilder(location.toString(), table)
+                        .withEnv(env)
+                        .withContinuousMode(isStreaming)
+                        .withPartitions(specifiedPartition)
+                        .build();
+        new CompactorSinkBuilder(table).withInput(source).build();
+        JobClient client = env.executeAsync("Compaction Job: " + location);
```

Comment:
Just use `execute(String jobName)`? Let `execution.attached` to decide whether to wait the execution.